### DISCUSSION TITLE
How to get default value from multidimensional dictionary

### DISCUSSION THREAD

**Discussion by vladimirs-git** [netbox-community/netbox#10885] [General]
For example, I'm processing Netbox objects, where some items have a reference to the site object and some don't.
In the case when I'm tying to get data from an object without a site, I raise TypeError, but I want to get an empty string.
What is the most elegant way to do this? My target to write simpler code without multiple data types checking.

```
items = [
    {'id': 1541,
     'prefix': '10.7.0.0/24',
     'site': {'display': 'SITE1',
              'id': 50,
              'name': 'SITE1',
              'slug': 'site1',
              'url': 'https://demo.netbox.dev/api/dcim/sites/50/'}},

    {'id': 1611,
     'prefix': '10.1.14.0/24',
     'site': None},
]

# very simple code to get site name, I like it ;)
print("site name: ", items[0]["site"]["name"])
# site name:  SITE1

# Instead of the following code I want to get empty string without data type validation, like in previous case
try:
    print("site name: ", items[1]["site"]["name"])
except TypeError:
    print("site absent")
    data = items[1].get("site") or {}
    value = data.get("name", "")
    print("site name: ", value)
# site absent
# site name:
```

**Comment by markkuleinio** [comment]
```
site_name = items[1].site.name if items[1].site else ""
print("Site name:", site_name)
```

Oops sorry I thought you had pynetbox objects, my bad. This works for you:

```
site_name = items[1]["site"]["name"] if items[1]["site"] else ""
print("Site name:", site_name)
```

**Comment by vladimirs-git** [comment]
Thank You [markkuleinio](https://github.com/markkuleinio)!
I have one more question. 
In this example items[x]["site"] data type can be 'dict' or 'None' and never 'list' or 'str'. 
Is it right?
Is there a way to be sure while expecting data types, some like API Schema?

**Reply by markkuleinio** [reply]
> In this example items[x]["site"] data type can be 'dict' or 'None' and never 'list' or 'str'.

That's the assumption in this example, yes.

If you don't know exactly what data type to expect, then you need to check it yourself, like by using `isinstance(item, some_type)` or by catching the exception as you did.

**Reply by vladimirs-git** [reply]
Many thanks!
